game_engine.py

```diff
@@ -0,0 +1,242 @@
+import json
+import sys
+import shutil
+import os
+import atexit
+
+
+class Game:
+    def __init__(self, map_file, game_window):
+        self.map_file = map_file
+        self.data = self.load_map(map_file)
+        self.game_window = game_window
+
+        if not self.data:
+            sys.exit('Error loading map file.')
```

Comment:
sys.exit is not standard for Sugar activities, but as it indicates a software configuration issue that shouldn't be a problem.

Reply:
So I leave it as it is?

Reply:
My role is review, not gatekeeper.  I see something unusual, I'll call it out.  Calling sys.exit is not standard; the outcome may not be as predictable as the more normal termination methods, because exit handlers will be called in an order defined by their setup in the Sugar toolkit.  We don't generally test the use of sys.exit, so if a fault is present or does develop in the Sugar toolkit your activity may be affected.  The usual way to exit is to call an Activity class method, which triggers the GTK main loop to exit, and whatever called that loop regains control and calls sys.exit eventually.